tools/publishIsolate.ts

```diff
@@ -0,0 +1,73 @@
+import fs from 'fs';
+import path from 'path';
+import childProcess from 'child_process';
+import getNpmPackageVersion from 'get-npm-package-version';
+import { compareVersions } from 'compare-versions';
+import isolateBuildConfig from '../conf/isolate.config.json';
+
+async function main() {
+  const isolatePackageRoot = './dist/isolate';
+
+  const publishPackageList: { packageName: string; dir: string, version: string }[] = [];
+
+  const configPackages = isolateBuildConfig.builds
+    .map(build => ({ buildName: build.name, version: build.version }));
+
+  for (const configPackage of configPackages) {
+    const filePath = path.join(isolatePackageRoot, configPackage.buildName);
+    if (!fs.existsSync(filePath)) {
+      throw new Error(`Package ${configPackage.buildName} does not exist`);
+    }
+    const stat = fs.statSync(filePath);
+    if (stat.isDirectory()) {
+      // get package.json
+      const packageJsonPath = path.join(filePath, 'package.json');
+      const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
+      // check is builded
+      if (packageJson.version !== configPackage.version) {
+        throw new Error(`Package ${configPackage.buildName} is not builded`);
+      }
+      // get version
+      const version = getNpmPackageVersion(packageJson.name) || '1.0.0';
```

Comment:
Get version before check directory and build it.

Reply:
Do you mean to put the command to build?

Reply:
Yes.

- get npm version
- check version
- yarn build:isolate:xxx
- yarn publish